examples/confetti_snapshot_animation/animated_explosion/mod.rs

```diff
@@ -0,0 +1,43 @@
+mod imp;
+
+use crate::{Explosion, ExplosionParameters};
+use glib::subclass::prelude::*;
+use gtk::prelude::*;
+use gtk::{gdk, glib, graphene};
+
+glib::wrapper! {
+    pub struct AnimatedExplosion(ObjectSubclass<imp::AnimatedExplosion>);
+}
+
+impl AnimatedExplosion {
+    pub(super) fn new(parameters: ExplosionParameters) -> Self {
+        let this: Self = glib::Object::new(&[]).expect("Failed to create a AnimatedExplosion");
+        let imp = this.imp();
+        imp.explosion.replace(Some(Explosion::new(parameters)));
+        this
+    }
+    pub(super) fn init_time(&self, clock: &gdk::FrameClock, duration: f64) {
```

Comment:
What is the point of having a public function that calls a private one? I would just have the implementation here.